### ISSUE TITLE
OneInstancePerTest trait only displays "in" test name, ignoring ShouldVerb, WhenVerb, etc..

### ISSUE THREAD

**Issue by GoogleCodeExporter** [seratch/scalatest#12]
```
What steps will reproduce the problem?
1. Create simple test with hierarchy using ShouldVerb, WhenVerb and InVerb
2. Observe test name hierarchy is correct
3. Mix in OneInstancePerTest trait

e.g.
class WithOneInstancePerTest extends WordSpec with ShouldMatchers with 
OneInstancePerTest  {
  "foo" should {
    "throw XYZ" when {
      "abc is called" in {
        true should equal(true)
      }
    }
  }
}

class WithoutOneInstancePerTest extends WordSpec with ShouldMatchers  {
  "foo" should {
    "throw XYZ" when {
      "abc is called" in {
        true should equal(true)
      }
    }
  }
}

Expected:
[info] WithoutOneInstancePerTest:
[info] foo 
[info]   should throw XYZ 
[info]   - when abc is called

Actual:
[info] WithOneInstancePerTest:
[info]   - when abc is called

Using sbt version 0.11.3, Scala 2.9.1 and ScalaTest 1.7.2
```

Original issue reported on code.google.com by `richardw...@googlemail.com` on 19 Jul 2012 at 2:23

**Comment by GoogleCodeExporter** [comment]
```
This is fixed in ScalaTest trunk already, and will be included in the soon to 
be release 2.0-M2.
```

Original comment by `b...@artima.com` on 19 Jul 2012 at 2:29

**Comment by GoogleCodeExporter** [comment]
```
Actually, the fix is already released in 2.0-SNAPSHOT.
```

Original comment by `b...@artima.com` on 19 Jul 2012 at 3:07

**Comment by GoogleCodeExporter** [comment]
Original comment by `billvenn...@gmail.com` on 19 Jul 2012 at 3:09
- Changed state: **Verified**
